pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,174 @@
+
+		// We use nodeInformer for observing current worker latency profile
+		nodeInformer.Informer(),
+
+		// for configmaps of operator client target namespace
+		kubeInformersForNamespaces.InformersFor(targetNamespace).Core().V1().ConfigMaps().Informer(),
+	).ResyncEvery(time.Minute).WithSync(ret.sync).WithSyncDegradedOnError(operatorClient).ToController(
+		"WorkerLatencyProfile",
+		eventRecorder.WithComponentSuffix("latency-profile-controller"),
+	)
+}
+
+func (c *LatencyProfileController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+
+	// Collect the current latency profile
+	configNodeObj, err := c.configNodeLister.Get("cluster")
+
+	// if config/v1/node/cluster object is not found this controller should do nothing
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Warningf("nodes.config.openshift.io/cluster not found")
+			return nil
+		}
+		return err
+	}
+
+	// in case of empty workerlatency profile, set status Complete=False and pre-empt sync
+	if configNodeObj.Spec.WorkerLatencyProfile == "" {
+		// TODO: in case worker latency profile is transitioned
+		// from Default/Medium/Low to ""
+		// check that the required args should vanish
+		_, err = c.updateStatus(ctx, reasonLatencyProfileEmpty, "latency profile not set on cluster", false, true)
+		return err
+	}
+
+	_, operatorStatus, _, err := c.operatorClient.GetStaticPodOperatorState()
+	if err != nil {
+		return err
+	}
+
+	// Collect the unique set of revisions for all nodes in cluster
+	revisionMap := map[int32]struct{}{}
+	uniqueRevisions := []int32{}
+	for _, nodeStatus := range operatorStatus.NodeStatuses {
+		revision := nodeStatus.CurrentRevision
+		if _, ok := revisionMap[revision]; !ok {
+			revisionMap[revision] = struct{}{}
+			uniqueRevisions = append(uniqueRevisions, revision)
+		}
+	}
+
+	// For each revision, check that the configmap for that revision have correct arg val pairs or not
+	revisionsHaveSynced, err := c.matchRevisionsFn(configNodeObj.Spec.WorkerLatencyProfile, uniqueRevisions)
+	if err != nil {
+		return err
+	}
+
+	if revisionsHaveSynced {
+		_, err = c.updateStatus(ctx, reasonLatencyProfileUpdated, "all static pod revision(s) have updated latency profile", false, true)
+	} else {
+		_, err = c.updateStatus(ctx, reasonLatencyProfileUpdateTriggered, "one or more static pod revision(s) are updating latency profile", true, false)
+	}
+	return err
+}
+
+func (c *LatencyProfileController) updateStatus(ctx context.Context, reason, message string, progressing, complete bool) (bool, error) {
+	progressingCondition := operatorv1.OperatorCondition{
+		Type:   wlpPrefix + "Progressing",
+		Status: operatorv1.ConditionUnknown,
+	}
+	completedCondition := operatorv1.OperatorCondition{
+		Type:   wlpPrefix + "Complete",
+		Status: operatorv1.ConditionUnknown,
+	}
+
+	if progressing {
+		progressingCondition.Status = operatorv1.ConditionTrue
+		progressingCondition.Message = message
+		progressingCondition.Reason = reason
```

Comment:
this is hard to guess from the function header. Can we have `progressingReason` and `completedReason` in there? and pass for example also `"AsExpected"` accordingly.

Reply:
although I feel `"AsExpected"` could be named better for each state and tailored to each condition.

Reply:
Updated function headers.

Reply:
Contd. in https://github.com/openshift/library-go/pull/1364/#discussion_r870361860.